octree_web_viewer/src/backend.rs

```diff
@@ -65,82 +77,105 @@ fn pad(input: &mut Vec<u8>) {
+                let mut reply_blob = Vec::<u8>::new();
+
+                let mut num_nodes_fetched = 0;
+                let mut num_points = 0;
+                for node_id in nodes_to_load {
+                    let mut node_data = octree
+                        .get_node_data(&node_id)
+                        .map_err(|_error| {
+                            crate::backend_error::PointsViewerError::NotFound(format!(
+                                "Could not get node {}.",
+                                node_id
+                            ))
+                        })
+                        .unwrap();
+
+                    // Write the bounding box information.
+                    let min = node_data.meta.bounding_cube.min();
+                    reply_blob.write_f32::<LittleEndian>(min.x).unwrap();
+                    reply_blob.write_f32::<LittleEndian>(min.y).unwrap();
+                    reply_blob.write_f32::<LittleEndian>(min.z).unwrap();
+                    reply_blob
+                        .write_f32::<LittleEndian>(node_data.meta.bounding_cube.edge_length())
+                        .unwrap();
+
+                    // Number of points.
+                    reply_blob
+                        .write_u32::<LittleEndian>(node_data.meta.num_points as u32)
+                        .unwrap();
+
+                    // Position encoding.
+                    let bytes_per_coordinate =
+                        node_data.meta.position_encoding.bytes_per_coordinate();
+                    reply_blob.write_u8(bytes_per_coordinate as u8).unwrap();
+                    assert!(
+                        bytes_per_coordinate * node_data.meta.num_points as usize * 3
+                            == node_data.position.len()
+                    );
+                    assert!(node_data.meta.num_points as usize * 3 == node_data.color.len());
+                    pad(&mut reply_blob);
+
+                    reply_blob.append(&mut node_data.position);
+                    pad(&mut reply_blob);
+
+                    reply_blob.append(&mut node_data.color);
+                    pad(&mut reply_blob);
+
+                    num_nodes_fetched += 1;
+                    num_points += node_data.meta.num_points;
+                }
+
+                let duration_ms = (time::precise_time_ns() - start) as f32 / 1000000.;
+                println!(
+                    "Got {} nodes with {} points ({}ms).",
+                    num_nodes_fetched, num_points, duration_ms
+                );
+
+                result(Ok(HttpResponse::Ok()
+                    .content_type("application/octet-stream")
+                    // disabling default encoding:
+                    // Local test (same machine) default encoding doubles the computing time in that condition by saving only 10% of the data volume
+                    // TODO tests are required to find the most meaningful option
```

Comment:
We usually write TODOs in the following way:
```
// TODO(author): ...
```
So your comment line would become:
```
// TODO(cvitadello): More tests are required in production to determine the best encoding.
```